### DISCUSSION TITLE
About RJSON and using a WebAssembly based parser

### DISCUSSION THREAD

**Discussion by Himujjal** [rome/tools#1410] [Suggestions]
Co-incidently I wrote another type of JSON superset named [EKON](https://github.com/ekon-org/ekon#syntax-and-specification) which is almost similar to RJSON. I wrote a super-fast parser for the same in C.  Well tested.

Will it be viable to use a WASM version of it? I can work on that. Its almost done. A few tweaks are required

I am considering disabandoning the EKON project and rename it as the RJSON if the rome guys are alright with it. Also maybe host RJSON as a separate repository in rome github organization.

What do you say?

**Comment by sebmck** [comment]
Hey! We're actually removing RJSON (per #1081), I have a WIP branch with it removed and a fully specced TOML parser instead. The biggest issue with these custom formats is not being able to parse and serialize them in other languages, and for more esoteric use-cases and companies is really important.

**Reply by Himujjal** [reply]
No. Don't do it. Please. I will maintain the parsers for other languages. I already know how to write the parser in C. It would be a cake in other languages. I think RJSON is a leap forward. Plus TOML sucks. Its very non-intuitive.

One thing that's significant is matching the indentation and comments while deserializing and serializing which is quite necessary for configs.

Plus also checkout [tree-sitter-ekon](https://github.com/ekon-org/tree-sitter-ekon) which is a nice cross-language parser for EKON (can be modified for RJSON).

**Reply by sebmck** [reply]
It's just a configuration language and at the end of the day we need to decrease barriers to adoption, and a completely new format is intimidating for users.

We already encourage configuration to be updated via the `rome config` command, so adopting a standardized and well-supported configuration format that's objectively non-intuitive isn't a problem.

**Reply by Himujjal** [reply]
I mean, you could have set the precedent. Rome is a big project. YAML is something of a curse for configuration and a lot of dev time has been wasted in fixing YAML.

RJSON could have put RJSON as YAML alternative in public eyes. That's what I think was more important. TOML is good enough but as I said. Learning TOML is more difficult than RJSON if barrier is concerned.

**Reply by Himujjal** [reply]
Anyways. If you think its not worth it. Its your project. I am fine with whatever you end up having a go at.
